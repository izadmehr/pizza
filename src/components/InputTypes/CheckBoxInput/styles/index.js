import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
`;
export const TextInput = styled.input`
  border: none;
  font-size: 1rem;
  padding: 0 5px;
  background-color: transparent;
  caret-color: ${props => props.theme.color.primary};

  ::-ms-clear {
    display: none;
  }

  ::-webkit-input-placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }

  ::placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }
  :focus {
    outline: none;
  }
`;
