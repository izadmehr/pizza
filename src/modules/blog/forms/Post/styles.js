import styled from "styled-components";

import { H3 } from "../../../../components/Typography";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
