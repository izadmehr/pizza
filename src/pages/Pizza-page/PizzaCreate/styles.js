import styled from "styled-components";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;
