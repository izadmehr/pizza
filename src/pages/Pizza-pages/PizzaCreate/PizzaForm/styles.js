import styled from "styled-components";

import { H3 } from "components/Typography";

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
