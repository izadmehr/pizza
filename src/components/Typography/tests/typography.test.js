import React from "react";
import { mount } from "enzyme";

import { theme } from "styles";

import "jsdom-global/register";
import "jest-styled-components";
import { H1, H2, H3, P, HR, A } from "../index";

describe("Typography Components", () => {
  it("H1 has correct default styles", () => {
    const wrapper = mount(<H1 theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("H2 has correct default styles", () => {
    const wrapper = mount(<H2 theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("H3 has correct default styles", () => {
    const wrapper = mount(<H3 theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("P has correct default styles", () => {
    const wrapper = mount(<P theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("HR has correct default styles", () => {
    const wrapper = mount(<HR theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("A has correct default styles", () => {
    const wrapper = mount(<A theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
