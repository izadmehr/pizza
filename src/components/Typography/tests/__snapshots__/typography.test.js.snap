// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Components A has correct default styles 1`] = `
.c0 {
  color: #4a89dc;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #5d9cec;
}

.c0:active,
.c0:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<styled.a
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  color: ",
            [Function],
            ";
  text-decoration: none;

  &:hover {
    text-decoration: none;
    color: ",
            [Function],
            ";
  }

  &:active,
  &:focus {
    outline: none;
    text-decoration: none;
  }
",
          ],
        },
        "displayName": "styled.a",
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "a",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <a
      className="c0"
    />
  </StyledComponent>
</styled.a>
`;

exports[`Typography Components H1 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 21px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 24px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 36px;
    line-height: 48px;
  }
}

<styled.h1
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 21px;
  line-height: 24px;

  &::selection {
    background: ",
            [Function],
            ";
  }

  ",
            "@media (min-width: 34.375rem)",
            " {
    font-size: 24px;
    line-height: 36px;
  }

  ",
            "@media (min-width: 48rem)",
            " {
    font-size: 36px;
    line-height: 48px;
  }
",
          ],
        },
        "displayName": "styled.h1",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h1",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <h1
      className="c0"
    />
  </StyledComponent>
</styled.h1>
`;

exports[`Typography Components H2 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 18px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 21px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 24px;
    line-height: 24px;
  }
}

<styled.h2
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 18px;
  line-height: 24px;

  &::selection {
    background: ",
            [Function],
            ";
  }

  ",
            "@media (min-width: 34.375rem)",
            " {
    font-size: 21px;
    line-height: 36px;
  }

  ",
            "@media (min-width: 48rem)",
            " {
    font-size: 24px;
    line-height: 24px;
  }
",
          ],
        },
        "displayName": "styled.h2",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <h2
      className="c0"
    />
  </StyledComponent>
</styled.h2>
`;

exports[`Typography Components H3 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 16px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 18px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 21px;
    line-height: 24px;
  }
}

<styled.h3
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 16px;
  line-height: 24px;

  &::selection {
    background: ",
            [Function],
            ";
  }

  ",
            "@media (min-width: 34.375rem)",
            " {
    font-size: 18px;
    line-height: 36px;
  }

  ",
            "@media (min-width: 48rem)",
            " {
    font-size: 21px;
    line-height: 24px;
  }
",
          ],
        },
        "displayName": "styled.h3",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "h3",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <h3
      className="c0"
    />
  </StyledComponent>
</styled.h3>
`;

exports[`Typography Components HR has correct default styles 1`] = `
.c0 {
  box-sizing: content-box;
  height: 0;
  display: block;
  unicode-bidi: isolate;
  overflow: hidden;
  margin-top: 25px;
  margin-bottom: 25px;
  border: 0;
  border-top: 1px solid #f2f2f4;
}

<styled.hr
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  box-sizing: content-box;
  height: 0;
  display: block;
  unicode-bidi: isolate;
  overflow: hidden;
  margin-top: 25px;
  margin-bottom: 25px;
  border: 0;
  border-top: 1px solid #f2f2f4;
",
          ],
        },
        "displayName": "styled.hr",
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "hr",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <hr
      className="c0"
    />
  </StyledComponent>
</styled.hr>
`;

exports[`Typography Components P has correct default styles 1`] = `
.c0 {
  font-weight: lighter;
  margin: 0;
  font-size: 16px;
  line-height: 1.5;
}

.c0::selection {
  background: #5d9cec;
}

<styled.p
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  font-weight: lighter;
  margin: 0;
  font-size: 16px;
  line-height: 1.5;

  &::selection {
    background: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.p",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <p
      className="c0"
    />
  </StyledComponent>
</styled.p>
`;
